src/pages/projectView/projectActivityPanel/screens/rewardSelection/FundingFormRewards.tsx

```diff
@@ -93,7 +105,18 @@ export const FundingFormRewards = ({ readOnly, onRewardClick }: Props) => {
           <VStack mt={1} padding="2px" width={"100%"}>
             {availableRewards.map((reward) => {
               const count = getRewardCount(reward.id)
-              const add = () => handleAdd(reward.id, count)
+              const add = () => {
+                const rewardStockRemaining = reward.maxClaimable ? reward.maxClaimable - reward.sold : 100;
+                if(rewardStockRemaining > count) {
+                  handleAdd(reward.id, count)
+                } else {
+                  toast({
+                    title: 'Reward Limit',
+                    description: `Maximum number of ${rewardStockRemaining} available for this reward`,
+                    status: 'error',
+                  })
+                }
+              }
```

Comment:
The above function (line 68-79) and this one, looks incredibly similar, maybe just add these to the `handleAdd` function itself, and pass in the whole `reward` instead of just `reward.id` , that way there wouldn't be repeated stuff.